cmd/thanos/sidecar.go

```diff
@@ -125,9 +125,14 @@ func runSidecar(
 
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
+			// Check prometheus's flags to ensure sane sidecar flags
+			err := checkFlags(ctx, logger, promURL, dataDir)
```

Comment:
Let's be more verbose and name this `validatePrometheus` because that what it is? (: 

Also can we put it in idiom:
`if err := ... ; ! err {`